channeldb/graph.go

```diff
@@ -3351,11 +3355,26 @@ func serializeChanEdgePolicy(w io.Writer, edge *ChannelEdgePolicy,
 		return err
 	}
 
+	// If the max_htlc field is present, we write it. To be compatible with
+	// older versions that wasn't aware of this field, we write it as part
+	// of the opaque data.
+	// TODO(halseth): clean up when moving to TLV.
+	var opaqueBuf bytes.Buffer
+	if edge.MessageFlags&lnwire.ChanUpdateOptionMaxHtlc != 0 {
+		err := binary.Write(&opaqueBuf, byteOrder, uint64(edge.MaxHTLC))
+		if err != nil {
+			return err
+		}
+	}
+
 	if len(edge.ExtraOpaqueData) > MaxAllowedExtraOpaqueBytes {
 		return ErrTooManyExtraOpaqueBytes(len(edge.ExtraOpaqueData))
 	}
+	if _, err := opaqueBuf.Write(edge.ExtraOpaqueData); err != nil {
```

Comment:
This write will cause the buf to expand and allocate more data. Why not just write the buf, then the opaque bytes which have already been allocated?

Reply:
Not sure I understand what you mean?

The rationale behind doing it this way is that we need to write the combined bytes of`max_htlc` + `opaque_bytes` as `VarBytes` to be compatible with the old format. Are you saying we should do something like 
```go
scratch [8]byte = bytes(max_htlc)
wire.WriteVarBytes(w, 0, append(scratch[:], edge.ExtraOpaqueData...))
```
?